schema/elements.go

```diff
@@ -16,9 +16,14 @@ limitations under the License.
 
 package schema
 
+import "sync"
+
 // Schema is a list of named types.
 type Schema struct {
 	Types []TypeDef `yaml:"types,omitempty"`
+
+	once sync.Once
+	m    map[string]TypeDef
```

Comment:
I'm not super excited about this, it means it's no longer OK to modify a schema if ?some? function has been called on it.

Reply:
All of this would be const/final if we could, there is no reason to change the schema while processing things?

Reply:
At a minimum, you need to make that clear in the documentation on these types.